pkg/apis/operator/v1/types.go

```diff
@@ -91,6 +92,11 @@ type InstallationSpec struct {
 	// kubernetesProvider.
 	// +optional
 	FlexVolumePath string `json:"flexVolumePath,omitempty"`
+
+	// MaxUnavailable determines the max number of daemonset pods that can be unavailable during an update.
+	// Default: 1
+	// +optional
+	MaxUnavailable *intstr.IntOrString `json:"maxUnavailable,omitempty"`
```

Comment:
Wondering if we should expose the full update strategy section wholesale: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.17/#daemonsetupdatestrategy-v1-apps

That would give us some more flexibility if we want to let users customize this further. 

I think we also need to consider what this means - there are multiple deployments / daemonsets / etc that this field could apply to across the operator, and it's not obvious from the name / comment which one this applies to.

I think the intention is that this only applies to the node daemonset?

Reply:
Do we really want to allow the user to switch the type to "OnDelete"?
And what is the implication for the namespace migration?

Reply:
@tmjd good question. I think probably not generically, but it could be useful for debugging, etc.

I suspect we should be validating that field to make sure it is set to "RollingUpdate", since that's the only mode that we support / test right now. I do think it still makes sense to maintain the structure, in case we want to support other modes in the future, or if there are additional fields added to the rolling update strategy that we want to enable. WDYT?

Reply:
For namespace migration, we probably only care about the maxUnavailable field.